From b131cf2c804436d28836a404ee9b23f659114d20 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Thu, 15 Aug 2019 14:21:23 +0800
Subject: [PATCH] arch:arm64:dts: add reg property to reserve memory statically

When reserve memory with no-map preproty dynamically, it causes
memory leak cheching feature works fail. So, It is need to reserve
memory statically if there is no-map preproty in dts node.
Therefore, remove alloc-ranges and size properties and then add
reg property so that change the memory reserving from dynamically
into statically

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts
index c1a030dca4c1..456c4fea280f 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts
@@ -43,16 +43,14 @@
 		/* device specific region for Lossy Decompression */
 		lossy_decompress: linux,lossy_decompress@54000000 {
 			no-map;
-			alloc-ranges = <0x0 0x6d000000 0x0 0x03000000>;
-			size = <0x0 0x03000000>;
+			reg = <0x00000000 0x6d000000 0x0 0x03000000>;
 		};
 
 		/* global autoconfigured region for contiguous allocations */
 		linux,cma@57000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			alloc-ranges = <0x0 0x70000000 0x0 0x8000000>;
-			size = <0x0 0x8000000>;
+			reg = <0x00000000 0x70000000 0x0 0x8000000>;
 			linux,cma-default;
 		};
 
@@ -60,8 +58,7 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			alloc-ranges = <0x0 0x78000000 0x0 0x8000000>;
-			size = <0x0 0x8000000>;
+			reg = <0x00000000 0x78000000 0x0 0x8000000>;
 		};
 	};
 
-- 
2.17.1

